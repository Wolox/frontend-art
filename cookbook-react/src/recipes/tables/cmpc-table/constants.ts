export const data = [
  {
    id: 1,
    data: 'Sample1'
  },
  {
    id: 2,
    data: 'Sample2'
  },
  {
    id: 3,
    data: 'Sample3'
  },
  {
    id: 4,
    data: 'Sample4'
  }
];
