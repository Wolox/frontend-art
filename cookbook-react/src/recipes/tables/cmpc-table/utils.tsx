import React from 'react';

import styles from './styles.module.scss';

/* eslint-disable react/no-multi-comp */

export const getColumns = () => [
  {
    title: 'Column1',
    render: (rowData: any) => <span className="text-big"> {rowData.id}</span>,
    className: styles.orderColumn
  },
  {
    title: 'Column2',
    render: (rowData: any) => <span className="text-big"> {rowData.data}</span>,
    className: styles.dateColumn
  },
  {
    title: 'Column3',
    render: (rowData: any) => <span className="text-big underline"> {rowData.data}</span>,
    className: styles.quantityColumn
  },
  {
    title: 'Column4',
    render: (rowData: any) => <span className="text-big underline">{rowData.data}</span>,
    className: styles.measureColumn
  },
  {
    title: 'Column5',
    render: (rowData: any) => <span className="text-big underline">{rowData.data}</span>,
    className: styles.measureColumn
  }
];

/* eslint-enable react/no-multi-comp */
