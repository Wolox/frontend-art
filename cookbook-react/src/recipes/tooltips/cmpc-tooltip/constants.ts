export const COLORS = {
  alabaster: '#F7F7F7',
  boulder: '#767676'
}