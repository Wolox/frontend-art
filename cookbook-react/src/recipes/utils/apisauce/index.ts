import { create, ApiResponse } from 'apisauce';

import { CamelcaseSerializer, SnakecaseSerializer } from 'cerealizr';

const deserializer = new CamelcaseSerializer();
const serializer = new SnakecaseSerializer();

export const STATUS_CODES = {
  unauthorized: 401
};

const api = create({
  baseURL: process.env.REACT_API_BASE_URL,
  timeout: 15000
});

// If you need to add more monitors consider calling api.addMonitor from your component
// eslint-disable-next-line no-unused-vars, prettier/prettier, @typescript-eslint/no-unused-vars
export const apiSetup = (onResponse?: (res: ApiResponse<any>) => void) => {
  api.addMonitor(response => {
    onResponse?.(response);
  });

  api.addResponseTransform(response => {
    if (response.data) {
      response.data = deserializer.serialize(response.data);
    }
  });

  api.addRequestTransform(request => {
    if (request.data) {
      request.data = serializer.serialize(request.data);
    }
  });
};

export default api;
