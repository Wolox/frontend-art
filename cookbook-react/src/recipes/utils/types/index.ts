// Useful basic types
export type Nullable<T> = T | null;

// Id/Item/Option types
export type Id = string | number;

export interface ItemType<IdType extends Id> {
  id: IdType;
}
