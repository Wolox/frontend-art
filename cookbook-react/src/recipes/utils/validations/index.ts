const emailRegex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;

const numericalDigitsRegex = /^([0-9]*)$/;

export const required = (errorMessage: string) => (val?: any) => (val ? undefined : errorMessage);

export const minLength = (length: number, errorMessage: string) => (val?: string) =>
  val && val.length >= length ? undefined : errorMessage;

export const maxLength = (length: number, errorMessage: string) => (val?: string) =>
  val && val.length <= length ? undefined : errorMessage;

export const pattern = (patternVal: RegExp, errorMessage: string) => (val?: string) =>
  val && patternVal.test(val) ? undefined : errorMessage;

export const email = (errorMessage: string) => pattern(emailRegex, errorMessage);

export const numericalDigits = (errorMessage: string) => pattern(numericalDigitsRegex, errorMessage);
