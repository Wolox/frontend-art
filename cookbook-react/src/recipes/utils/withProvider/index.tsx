import React, { useReducer } from 'react';

interface ActionType {
  type: string;
}

interface Props<S extends {}, A> {
  Context: React.Context<{ state: S; dispatch: React.Dispatch<A> }>;
  reducer: React.Reducer<S, A>;
  initialState: S;
}

const withProvider = <P extends {}, S, A extends ActionType>({
  Context,
  reducer,
  initialState
}: Props<S, A>) => (WrappedComponent: React.ComponentType<P>) => {
  function ProviderWrapper(props: P) {
    const [state, dispatch] = useReducer(reducer, initialState);
    return (
      <Context.Provider value={{ state, dispatch }}>
        <WrappedComponent {...props} />
      </Context.Provider>
    );
  }

  return ProviderWrapper;
};

export default withProvider;
