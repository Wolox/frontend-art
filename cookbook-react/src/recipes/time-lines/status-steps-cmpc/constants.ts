import { StatusType } from './types';

export const statusNames = {
  [StatusType.CREATED]: 'Creado',
  [StatusType.CONFIRMED]: 'Confirmado',
  [StatusType.IN_PRODUCTION]: 'En producci√≥n',
  [StatusType.DELIVERY]: 'En despacho',
  [StatusType.DELIVERED]: 'Entregado',
  [StatusType.CLOSED]: 'Finalizado'
};
