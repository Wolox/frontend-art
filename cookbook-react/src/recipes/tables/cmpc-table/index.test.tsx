import React from 'react';
import { render } from '@testing-library/react';

import { getColumns } from './utils';
import { data } from './constants';

import Table from './index';

describe('Table component', () => {
  it('renders without crashing', () => {
    const { asFragment } = render(<Table columns={getColumns()} data={data} emptyMessage="" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
