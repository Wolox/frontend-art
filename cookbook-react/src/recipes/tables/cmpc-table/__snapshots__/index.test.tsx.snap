// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="tableContainer"
  >
    <div
      class="row space-between tableRowHeader"
    >
      <div
        class="row center middle tableHeader orderColumn"
      >
        Column1
      </div>
      <div
        class="row center middle tableHeader dateColumn"
      >
        Column2
      </div>
      <div
        class="row center middle tableHeader quantityColumn"
      >
        Column3
      </div>
      <div
        class="row center middle tableHeader measureColumn"
      >
        Column4
      </div>
      <div
        class="row center middle tableHeader measureColumn"
      >
        Column5
      </div>
    </div>
    <div
      class="row space-between tableRowBody"
    >
      <div
        class="row center middle text-big tableCell orderColumn"
      >
        <span
          class="text-big"
        >
           1
        </span>
      </div>
      <div
        class="row center middle text-big tableCell dateColumn"
      >
        <span
          class="text-big"
        >
           Sample1
        </span>
      </div>
      <div
        class="row center middle text-big tableCell quantityColumn"
      >
        <span
          class="text-big underline"
        >
           Sample1
        </span>
      </div>
      <div
        class="row center middle text-big tableCell measureColumn"
      >
        <span
          class="text-big underline"
        >
          Sample1
        </span>
      </div>
      <div
        class="row center middle text-big tableCell measureColumn"
      >
        <span
          class="text-big underline"
        >
          Sample1
        </span>
      </div>
    </div>
    <div
      class="row space-between tableRowBody"
    >
      <div
        class="row center middle text-big tableCell orderColumn"
      >
        <span
          class="text-big"
        >
           2
        </span>
      </div>
      <div
        class="row center middle text-big tableCell dateColumn"
      >
        <span
          class="text-big"
        >
           Sample2
        </span>
      </div>
      <div
        class="row center middle text-big tableCell quantityColumn"
      >
        <span
          class="text-big underline"
        >
           Sample2
        </span>
      </div>
      <div
        class="row center middle text-big tableCell measureColumn"
      >
        <span
          class="text-big underline"
        >
          Sample2
        </span>
      </div>
      <div
        class="row center middle text-big tableCell measureColumn"
      >
        <span
          class="text-big underline"
        >
          Sample2
        </span>
      </div>
    </div>
    <div
      class="row space-between tableRowBody"
    >
      <div
        class="row center middle text-big tableCell orderColumn"
      >
        <span
          class="text-big"
        >
           3
        </span>
      </div>
      <div
        class="row center middle text-big tableCell dateColumn"
      >
        <span
          class="text-big"
        >
           Sample3
        </span>
      </div>
      <div
        class="row center middle text-big tableCell quantityColumn"
      >
        <span
          class="text-big underline"
        >
           Sample3
        </span>
      </div>
      <div
        class="row center middle text-big tableCell measureColumn"
      >
        <span
          class="text-big underline"
        >
          Sample3
        </span>
      </div>
      <div
        class="row center middle text-big tableCell measureColumn"
      >
        <span
          class="text-big underline"
        >
          Sample3
        </span>
      </div>
    </div>
    <div
      class="row space-between tableRowBody"
    >
      <div
        class="row center middle text-big tableCell orderColumn"
      >
        <span
          class="text-big"
        >
           4
        </span>
      </div>
      <div
        class="row center middle text-big tableCell dateColumn"
      >
        <span
          class="text-big"
        >
           Sample4
        </span>
      </div>
      <div
        class="row center middle text-big tableCell quantityColumn"
      >
        <span
          class="text-big underline"
        >
           Sample4
        </span>
      </div>
      <div
        class="row center middle text-big tableCell measureColumn"
      >
        <span
          class="text-big underline"
        >
          Sample4
        </span>
      </div>
      <div
        class="row center middle text-big tableCell measureColumn"
      >
        <span
          class="text-big underline"
        >
          Sample4
        </span>
      </div>
    </div>
  </div>
  <div
    class="row end middle m-top-5"
  >
    <div
      class="row middle center arrowIconContainer disabled"
    >
      <button
        disabled=""
        type="button"
      >
        &lt;
      </button>
    </div>
    <p
      class="text-big m-left-3 bold"
    >
      1Â 
    </p>
    <p
      class="tablePaginationTotal text-big m-right-3"
    >
      - 10
    </p>
    <div
      class="row middle center arrowIconContainer pointer"
    >
      <button
        type="button"
      >
        &gt;
      </button>
    </div>
  </div>
</DocumentFragment>
`;
